libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,226 @@
+import {
+    AfterViewInit,
+    ChangeDetectorRef,
+    ChangeDetectionStrategy,
+    Component,
+    EventEmitter,
+    Input,
+    Optional,
+    Output,
+    Self,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { CheckboxComponent as CoreCheckboxComponent } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+
+/**
+ * Checkbox implementation based on the
+ * https://github.com/SAP/fundamental-ngx/wiki/Platform:-Checkbox-Component-Technical-Design
+ * documents.
+ *
+ *
+ */
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class CheckboxComponent extends BaseInput implements AfterViewInit {
+    /** set to true if binary checkbox */
+    @Input()
+    isBinary: boolean = false;
+
+    /** Sets label for checkbox. */
+    @Input()
+    label: string;
+
+    /**
+     * Attached to the aria-label attribute of the host element. In most cases, aria-labelledby will
+     * take precedence so this may be omitted.
+     */
+    @Input('aria-label')
+    ariaLabel: string = '';
+
+    /**
+     * Users can specify the `aria-labelledby` attribute which will be forwarded to the input element
+     */
+    @Input('aria-labelledby')
+    ariaLabelledby: string | null = null;
+
+    /** indeterminate state */
+    @Input()
+    tristate: boolean = false;
+
+    /** indeterminate state can be selected  */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /** value for checkbox, when it is not binary */
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+    set value(selectValue: any) {
+        if (selectValue !== this.value) {
+            // baseInput class is calling this.onchange(value),
+            // which causing Expression change error, so doing all steps here.
+            this._value = selectValue;
+            this.stateChanges.next('writeValue');
+            this._cd.markForCheck();
+        }
+    }
+
+    /** Emitting checkbox change event */
+    @Output()
+    change = new EventEmitter();
+
+    /** @hidden
+     * tracking checkbox current value
+     */
+    public checkboxValue: any;
+
+    /* @hidden
+     * reference of child checkbox implementation
+     */
+    @ViewChild(CoreCheckboxComponent)
+    private corecheckbox: CoreCheckboxComponent;
+
+    /* @hidden
+     * stores  formControl values
+     */
+    private multiSelectModel: any;
+
+    constructor(
+        protected _changeDetector: ChangeDetectorRef,
+        @Optional() @Self() public ngControl: NgControl,
+        @Optional() @Self() public ngForm: NgForm
+    ) {
+        super(_changeDetector, ngControl, ngForm);
+    }
+
+    /** ControlvalueAccessor */
+    writeValue(value: any): void {
+        if (value) {
```

Comment:
What if someone changes value fo `false` from outside? It won't be triggered

Reply:
Check if you can change value to `false` from outside (programatically from docs)

Reply:
removed writevalue if(value), because it will block control value=false.